### PR TITLE
Add a simple POC VDF and integrate into the drand protocol

### PR DESCRIPTION
**Pull request by rlan35**
go test succeeded.
Running golint...
golint passed.
Running goimports...
goimports passed.
Running go generate...
go generate succeeded; all generated files seem up to date.
All Checks Passed!!! :-)

### CODE REVIEW

File: consensus/consensus.go [hunk 1]
```diff
@@ -140,6 +143,29 @@ func (consensus *Consensus) UpdateConsensusID(consensusID uint32) {
 	}
 }
 
+// WaitForNewRandomness listens to the RndChannel to receive new VDF randomness.
+func (consensus *Consensus) WaitForNewRandomness() {
```
Comment (mikedoan) [inline consensus/consensus.go]: can we add randomness as a service? should be decoupled from calling from consensus service?

File: api/service/consensus/service.go [hunk 2]
```diff
@@ -25,6 +25,7 @@ func (s *Service) StartService() {
 	s.stopChan = make(chan struct{})
 	s.stoppedChan = make(chan struct{})
 	s.consensus.WaitForNewBlock(s.blockChannel, s.stopChan, s.stoppedChan, s.startChan)
+	s.consensus.WaitForNewRandomness()
```
Comment (mikedoan) [inline api/service/consensus/service.go]: should we decouple randomness as a separated service?

File: consensus/consensus_leader.go [hunk 3]
```diff
@@ -67,6 +64,15 @@ func (consensus *Consensus) WaitForNewBlock(blockChannel chan *types.Block, stop
 					// TODO: check validity of pRnd
 					newBlock.AddRandPreimage(binary.BigEndian.Uint32(pRnd))
 				}
+				rnd, blockHash, err := consensus.GetNextRnd()
+				if err != nil {
```
Comment (chaosma) [inline consensus/consensus_leader.go]: err == nil ?

File: crypto/vdf/vdf.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+func (vdf *VDF) GetOutputChannel() chan [32]byte {
+	return vdf.outputChan
+}
+
+// Execute runs the VDF until it's finished and put the result into output channel.
+func (vdf *VDF) Execute() {
+	vdf.finished = false
+	tempResult := vdf.input
+	for i := 0; i < vdf.difficulty; i++ {
+		tempResult = sha3.Sum256(tempResult[:])
```
Comment (chaosma) [inline crypto/vdf/vdf.go]: how long it takes to run one iteration of sha3 sum?

File: crypto/vdf/vdf.go [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+// GetOutputChannel returns the vdf output channel.
+func (vdf *VDF) GetOutputChannel() chan [32]byte {
+	return vdf.outputChan
+}
+
+// Execute runs the VDF until it's finished and put the result into output channel.
+func (vdf *VDF) Execute() {
+	vdf.finished = false
+	tempResult := vdf.input
+	for i := 0; i < vdf.difficulty; i++ {
```
Comment (chaosma) [inline crypto/vdf/vdf.go]: will this use all of the cpu?

File: consensus/consensus.go (hunk 1 above)
Comment (rlan35) [reply consensus/consensus.go]: Randomness is already a separate service. But consensus needs to wait for output from randomness service

File: consensus/consensus_leader.go (hunk 3 above)
Comment (rlan35) [reply consensus/consensus_leader.go]: opps, Done.

File: crypto/vdf/vdf.go (hunk 5 above)
Comment (rlan35) [reply crypto/vdf/vdf.go]: It will use one core I think.

**Pull request merged**